### PR TITLE
Privacy manifest

### LINKED ISSUE #489: Missing API declarations
We need to take care of this before 1st of May.

>  Alephium Wallet
Version 1.0.8
Build 75
Although submission for TestFlight review was successful, you may want to correct the following issues in your next submission for TestFlight review. Once you've corrected the issues, upload a new binary to App Store Connect.

> ITMS-91053: Missing API declaration - Your app’s code in the “Alephium” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryFileTimestamp. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.

> ITMS-91053: Missing API declaration - Your app’s code in the “Alephium” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryDiskSpace. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.

> ITMS-91053: Missing API declaration - Your app’s code in the “Alephium” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategorySystemBootTime. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.

> ITMS-91053: Missing API declaration - Your app’s code in the “Alephium” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryUserDefaults. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.

### PR DESCRIPTION
**Pull request by mvaivre**
~WIP: I get a white screen on iOS.~

~Series of errors in the native logs:~

~Invariant Violation: Failed to call into JavaScript module method RCTLog.logIfNoNativeHook(). Module has not been registered as callable. Bridgeless Mode: false. Registered callable JavaScript modules (n = 0): .
A frequent cause of the error is that the application entry file path is incorrect. This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native., js engine: hermes~

Edit: reanimated was the culprit. I had to downgrade it, to be investigated later. Hopefully this package doesn't need any of the privacy reasons. 

--- 

EDIT ILIAS: Closes #489

### CODE REVIEW

**Comment by nop33** [comment]
I am having issues getting passed this:

```
pnpm exec expo prebuild
Cannot read properties of undefined (reading 'extract')
✖ Failed to create the native directories
You may want to delete the ./ios and/or ./android directories before trying again.
```

What am I doing wrong?

File: apps/mobile-wallet/app.config.js [hunk 1]
```diff
@@ -42,6 +42,26 @@ export default {
...
+          },
+          {
+            NSPrivacyAccessedAPIType: 'NSPrivacyAccessedAPICategorySystemBootTime',
+            NSPrivacyAccessedAPITypeReasons: ['35F9.1']
+          },
+          {
+            NSPrivacyAccessedAPIType: 'NSPrivacyAccessedAPICategoryUserDefaults',
+            NSPrivacyAccessedAPITypeReasons: ['CA92.1']
+          }
+        ]
```
Comment (nop33) [inline apps/mobile-wallet/app.config.js:64]: Are there any resources on how you came to decide to add these?

File: apps/mobile-wallet/package.json [hunk 2]
```diff
@@ -44,23 +44,23 @@
     "buffer": "^6.0.3",
     "colord": "^2.9.3",
     "dayjs": "^1.10.7",
-    "expo": "50.0.7",
-    "expo-application": "~5.8.3",
+    "expo": "50.0.17",
```
Comment (nop33) [inline apps/mobile-wallet/package.json:47]: Are all the updated expo deps compatible with this expo version?

File: apps/mobile-wallet/app.config.js (hunk 1 above)
Comment (mvaivre) [reply apps/mobile-wallet/app.config.js:64]: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api

**Comment by mvaivre** [comment]
> Anyway, I get a diff, do you get it too?

Yes, [here](https://github.com/alephium/alephium-frontend/pull/534/commits/338811e1ce0765dbb7900fa5b3b828b631463af2). And I don't like that there's a local path in there..!

File: apps/mobile-wallet/app.config.js (hunk 1 above)
Comment (nop33) [reply apps/mobile-wallet/app.config.js:64]: I saw you had posted a link to an expo issue somewhere but I can't find it.

**Comment by nop33** [comment]
Oh shit, you're right. It doesn't make sense to have a local path. So many questions...

1. how do we avoid committing it in the future since it will always appear when running prebuild?
2. will the app work fine if we don't commit it?

File: apps/mobile-wallet/app.config.js (hunk 1 above)
Comment (nop33) [reply apps/mobile-wallet/app.config.js:64]: ah, it was in the slack canvas

**Comment by nop33** [comment]
What process did you follow to decide which `NSPrivacyAccessedAPITypeReasons` values to use? Reading the [Apple docs](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api) I wouldn't know which ones to chose for each `NSPrivacyAccessedAPIType` listed in the [email](https://github.com/alephium/alephium-frontend/issues/489#issue-2216536051) that Apple sent us.

I understand that they want us to include reasons for the following API categories:
- `NSPrivacyAccessedAPICategoryFileTimestamp`
- `NSPrivacyAccessedAPICategoryDiskSpace`
- `NSPrivacyAccessedAPICategorySystemBootTime`
- `NSPrivacyAccessedAPICategoryUserDefaults`

But I don't understand why the values `C617.1`, `E174.1`, `35F9.1` and `CA92.1` were chosen instead of any others ([same question](https://github.com/expo/expo/issues/27796#issuecomment-2015538202) as this guy). I need to do more investigation, but if you already know the reasons let me know to save time.

To be honest, it feels weird to me that our app is accessing file timestamps, disk space, system boot time and user defaults. I don't think our code does any of that. But it's probably our outdated deps that do. Updated versions of the deps should include privacy files themselves.

It would be useful to know which packages are the ones that require these new privacy declarations. Maybe at the end of the day we don't need them or we can update them so that we use their privacy files.

For example, I found out that `@react-native-async-storage/async-storage` recently added a privacy file to [version 1.23.0](https://github.com/react-native-async-storage/async-storage/releases). We are using this package. Maybe if we update it we don't need to manually the `NSPrivacyAccessedAPICategoryFileTimestamp` category.

**Comment by nop33** [comment]
In the [installation docs](https://react-native-async-storage.github.io/async-storage/docs/install/) of `@react-native-async-storage/async-storage` I read that it should be installed with:

```
npx expo install @react-native-async-storage/async-storage
```

So I tried all these options but in all cases I see that the wrong version of expo is being used:

```
pnpm expo install @react-native-async-storage/async-storage
pnpm exec expo install @react-native-async-storage/async-storage
pnpm dlx expo install @react-native-async-storage/async-storage
```

```
› Installing 1 SDK 50.0.0 compatible native module using pnpm
> pnpm add @react-native-async-storage/async-storage@1.21.0
```

From what I understand, `npx expo` refers to `@expo/cli`, not the legacy `expo-cli`:

> The modern [local Expo CLI](https://blog.expo.dev/the-new-expo-cli-f4250d8e3421) is included with the expo package and does not need to be installed separately. Run it with npx expo. Its source code lives in the [expo/expo](https://github.com/expo/expo/blob/main/packages/%40expo/cli) repo.
https://github.com/expo/expo-cli

When I do `pnpm exec expo --version` I get `0.17.10`. But in the repo expo/expo repo I see that the latest version is `0.18.4`
https://github.com/expo/expo/blob/main/packages/%40expo/cli/package.json

**Comment by nop33** [comment]
If we want to release this before the dw secrets, we need to change the base branch from `next` to `master`.

**Review by nop33 — APPROVED** [review]

**Pull request merged**